### DISCUSSION TITLE
Wrapped Exceptions contain no Entity

### DISCUSSION THREAD

**Discussion by flx5** [resteasy/resteasy#3169] [General]
Hi,

I've just noticed that wrapped Exceptions do not contain any entity. The comment seems to indicate that the entity should be saved:

https://github.com/resteasy/resteasy/blob/4d6161defd36a5da7687f9a75d3d6ab0250d8423/resteasy-client-api/src/main/java/org/jboss/resteasy/client/exception/WebApplicationExceptionWrapper.java#L125-L143

What is the expected behaviour for this? Should the user of the library unwrap the Exception or is this a bug in resteasy?

**Comment by jamezp** [comment]
Hi @flx5. This was done for security purposes, see [RESTEASY-2781](https://issues.redhat.com/browse/RESTEASY-2781). It was undone a bit with [RESTEASY-3096](https://issues.redhat.com/browse/RESTEASY-3096).

What version of RESTEasy are you using?

**Reply by flx5** [reply]
Hi, that was fast. 
Okay, so after reading through those issues I think it should be safe to use unwrap when parsing the exception body in my catch block?

The version is 3.11.4.

What confused me was the javadoc comment in that version, but rereading the current version that seems to have been fixed already :)

Sorry for the inconvenience.

Thank you for your help!

**Reply by jamezp** [reply]
Yes that should definitely be safe. And it's not a problem at all. We are happy to help :)
